### PR TITLE
server: answers vote API

### LINKED ISSUE #74: server: add voting API to Answers

### PR DESCRIPTION
**Pull request by AaronCQL**
## Features

- [x] Endpoint to get user's vote status for answers of a question
- [x] Endpoint to upvote answer 
- [x] Endpoint to downvote answer
- [x] Tests
- [x] Docs

## Misc

- Refactor: did some minor refactoring of types and variable names as well for readability
- Test: add full suite of tests for vote API (both questions and answers and all edge cases; refer to [vote controller test](https://github.com/CS3219-SE-Principles-and-Patterns/cs3219-ay2021-s1-project-2020-s1-g15/blob/server-vote-api/server/tests/controllers/votes.test.ts))

Closes #74

### CODE REVIEW

File: server/tests/routes/answers.test.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+    const firstAnswerId: string = firstAnswer._id.toHexString();
+    const secondAnswerId: string = secondAnswer._id.toHexString();
+    const thirdAnswerId: string = thirdAnswer._id.toHexString();
+
+    const urlQuery = {
+      answerIds: [firstAnswerId, secondAnswerId, thirdAnswerId],
+    };
+    const { body } = await request(server)
+      .get(`${API_ENDPOINT}/vote-status?${stringify(urlQuery)}`)
```
Comment (AaronCQL) [inline server/tests/routes/answers.test.ts:128]: @EugeneTeu for client, can refer to this part when calling the answer's `/vote-status` endpoint. Basic gist is the `stringify` should take in a object with a single key `answerIds`, which is an array of answer id strings.

Comment (EugeneTeu) [reply server/tests/routes/answers.test.ts:128]: ok, have a similar `util` in FE side.

**Review by EugeneTeu — APPROVED** [review]
LGTM
